drift-codec/src/main/java/com/facebook/drift/codec/internal/ProtocolReader.java

```diff
@@ -127,6 +128,18 @@ public ByteBuffer readBinaryField()
         return fieldValue;
     }
 
+    public ByteBuffer readBinaryField(byte[] buf, int offset, int length)
+            throws TException
+    {
+        if (!checkReadState(TType.STRING)) {
+            return null;
+        }
+        currentField = null;
+        ByteBuffer fieldValue = ((TBinaryProtocol) protocol).readBinary(buf, offset, length);
```

Comment:
Are we confident it won't be needed in other protocols? How about implementing it for other protocols as well?